04_snake_game/main.go

```diff
@@ -129,28 +154,38 @@ func main() {
 		}
 	}()
 
-	ticker := time.NewTicker(70 * time.Millisecond)
+	ticker := time.NewTicker(90 * time.Millisecond)
 	defer ticker.Stop()
 
-	// This is the main event loop.
 	for {
 		select {
 		case ev := <-eventQueue:
 			if ev.Type == termbox.EventKey {
 				switch ev.Key {
 				case termbox.KeyArrowDown:
-					g = moveDown(g)
+					g.dir = coord{0, 1}
 				case termbox.KeyArrowUp:
-					g = moveUp(g)
+					g.dir = coord{0, -1}
 				case termbox.KeyArrowLeft:
-					g = moveLeft(g)
+					g.dir = coord{-1, 0}
 				case termbox.KeyArrowRight:
-					g = moveRight(g)
+					g.dir = coord{1, 0}
 				case termbox.KeyEsc:
 					return
 				}
 			}
 		case <-ticker.C:
+			if checkb(g) {
+				termbox.Clear(termbox.ColorWhite, termbox.ColorBlue)
+				writeText(70, 5, `Game Over`, termbox.ColorWhite, termbox.ColorBlue)
+				termbox.Flush()
+				time.Sleep(2 * time.Second)
+				return
+			}
+			if checka(g) {
+				g.score++
+				newApple(g.fieldWidth, g.fieldHeight)
```

Comment:
There is a bug. `newApple` returns a value, but you are not assigning it anywhere.